src/modeler/AutoRest.Swagger/Validation/Core/JsonValidationLogListener.cs

```diff
@@ -14,7 +14,7 @@ public class JsonValidationLogListener : ILogListener
     {
         private readonly List<Dictionary<string, string>> rawMessageCollection = new List<Dictionary<string, string>>();
 
-        private readonly Regex resPathPattern = new Regex(@"/providers/(?<providerNamespace>[^{/]+)/((?<resourceType>[^{/]+)/)?");
+        private readonly Regex resPathPattern = new Regex(@"/providers/(?<providerNamespace>[^{/]+)((/(?<resource>[^{/]+)/)((?<resourceName>[^/]+)))+(/(?<resource>[^{/]+))");
```

Comment:
nit: mark this as static so we don't have to generate this object for every validation